src/methods/extract.jl

```diff
@@ -102,28 +104,37 @@ function _extract(T, A::RasterStackOrArray, ::Nothing, geoms::AbstractArray; nam
         return skipmissing ? collect(_skip_missing_rows(rows, _missingval_or_missing(A))) : collect(rows)
     end
 end
-function _extract(T, A::RasterStackOrArray, ::GI.AbstractFeatureTrait, feature; kw...)
+function _extract(::Type{T}, A::RasterStackOrArray, ::GI.AbstractFeatureTrait, feature; kw...) where T
     _extract(T, A, GI.geometry(feature); kw...)
 end
-function _extract(T, A::RasterStackOrArray, ::GI.FeatureCollectionTrait, fc; kw...)
+function _extract(::Type{T}, A::RasterStackOrArray, ::GI.FeatureCollectionTrait, fc; kw...) where T
     # Fall back to the Array/iterator method for feature collections
     _extract(T, A, [GI.geometry(f) for f in GI.getfeature(fc)]; kw...)
 end
-function _extract(T, A::RasterStackOrArray, ::GI.AbstractMultiPointTrait, geom; skipmissing=false, kw...)
+function _extract(::Type{T}, A::RasterStackOrArray, ::GI.AbstractMultiPointTrait, geom; 
+    skipmissing=false, 
+    kw...
+) where T
     rows = (_extract_point(T, A, g; kw...) for g in GI.getpoint(geom))
     return skipmissing ? collect(_skip_missing_rows(rows, _missingval_or_missing(A))) : collect(rows)
 end
-function _extract(T, A::RasterStackOrArray, ::GI.AbstractGeometryTrait, geom;
+function _extract(::Type{T}, A::RasterStackOrArray, t::GI.AbstractGeometryTrait, geom;
     names, skipmissing=false, kw...
-)
+) where T
     # Make a raster mask of the geometry
-    B = boolmask(geom; to=commondims(A, DEFAULT_POINT_ORDER), kw...)
+    template = view(A, Touches(GI.extent(geom)))
+    ods = otherdims(A, DEFAULT_POINT_ORDER)
+    if length(ods) > 0
+        template = view(template, map(d -> rebuild(d, firstindex(d)), ods)) 
+    end
+    B = boolmask(geom; to=template, kw...)
+    offset = CartesianIndex(map(first, parentindices(template)))
     # Add a row for each pixel that is `true` in the mask
-    rows = (_maybe_add_fields(T, A, _prop_nt(A, I, names), I) for I in CartesianIndices(B) if B[I])
+    rows = (_maybe_add_fields(T, A, _prop_nt(A, I + offset, names), I) for I in CartesianIndices(B) if B[I])
```

Comment:
the second `I` should also be `I + offset` I think?

Reply:
yep